### PR TITLE
THREESCALE-8535 OSD 4.12 support

### PR DESCRIPTION
**Pull request by MStokluska**
What's done so far
- bumped Go to 1.17
- bumped k8s deps to 0.24.3
- bumped client-go to 0.24.3
- bumped controller-runtime v0.12.2
- bumped controller-gen 0.9.2

4.12 support
- Replaced the v1beta1 policies with v1

Other changes to suit deps bumps
- slight refactor to reconcilers
- slight refactor to apimanager watches
- update to single unit test 
- update to some deprecated logr commands
- other changes supporting the dep bumps

Unfortunately, deps could not be bumped any further due to the fact that it would have clashed with other operators that RHOAM has to keep in line with.

TODO:
Some PR tests still failing as this would need to get merged to get them pass: https://github.com/openshift/release/pull/33016
On cluster verification of initial install and upgrade

### CODE REVIEW

**Comment by MStokluska** [comment]
@eguzki I would appreciate it if you could have a look at it, main areas I'm concerned with are:
- apimanager reconciler watcher update
- grafana bump to v4 and removal on the spec.name field (yet to do on cluster upgrade verification)

**Review by eguzki — CHANGES_REQUESTED** [review]
Awesome work :medal_military: 

Left some comments

This graphana major version bump needs to be analyzed. What happens when upgrading from earlier version and 3scale has grafana dashboards deployed? The grafana operator needs to be upgraded, before the 3scale operator I guess. Needs to be documented.

File: doc/prometheusrules/apicast.yaml [hunk 1]
```diff
@@ -17,25 +17,31 @@ spec:
         description: Job {{ $labels.job }} on {{ $labels.namespace }} is DOWN
         sop_url: https://github.com/3scale/3scale-Operations/blob/master/sops/alerts/prometheus_job_down.adoc
         summary: Job {{ $labels.job }} on {{ $labels.namespace }} is DOWN
-      expr: up{job=~".*/apicast-production|.*/apicast-staging",namespace="__NAMESPACE__"} == 0
+      expr: up{job=~".*/apicast-production|.*/apicast-staging",namespace="__NAMESPACE__"}
+        == 0
```
[the remark this answers is not in the document]
Comment (MStokluska) [reply doc/prometheusrules/apicast.yaml:21]: Hey, this was done in order to get the prometheus rules tests to pass. I am not sure as of yet why is this required to get the tests pass (what changes the expected format) but will dig deeper.

**Comment by MStokluska** [comment]
> update Makefile's `go-get-tool` method to use `go install` instead of `go get`. Maybe rename the method to `go-bin-install` ??

@eguzki this change actually causes the issue around kustomize
```
go install: sigs.k8s.io/kustomize/kustomize/v3@v3.5.4 (in sigs.k8s.io/kustomize/kustomize/v3@v3.5.4):
	The go.mod file for the module providing named packages contains one or
	more exclude directives. It must not contain directives that would cause
	it to be interpreted differently than if it were the main module.
make: *** [Makefile:102: /go/src/github.com/3scale/3scale-operator/bin/kustomize] Error 1

Exited with code exit status 2
```
Will look into this.

**Comment by MStokluska** [comment]
E2E fails due to it being tested on ocp 4.6, which does not have policy/v1 but policy/v1beta1. I suggest bumping ocp testing clusters to at least 4.8

**Comment by eguzki** [comment]
I suggest to bump also kustomize and controller-gen, 

kustomize v4.5.7
controller-gen v0.9.2

Aligned with the apicast operator [PR](https://github.com/3scale/apicast-operator/pull/180)

**Review by eguzki — DISMISSED** [review]

**Comment by eguzki** [comment]
LGTM, before merging, update `/doc/development.md` accordingly

**Review by eguzki — CHANGES_REQUESTED** [review]
update /doc/development.md accordingly

**Review by eguzki — APPROVED** [review]

**Pull request merged**